### PR TITLE
fix: add ListObjects request validations and cleanup validation errors

### LINKED ISSUE #365: ListObjects: Errors are being swallowed
Tested with: `v0.2.5`, ran through docker.
Tests:

```sh
# user empty
curl  -X POST 'http://localhost:8080/stores/01GJZNM5ETC4YK34ZZST97R39X/list-objects' \
-H "Content-Type: application/json" \
-d '{"user": "", "relation":"reader","type":"resource"}' | jq
# user invalid (note valid on schema 1.0)
curl  -X POST 'http://localhost:8080/stores/01GJZNM5ETC4YK34ZZST97R39X/list-objects' \
-H "Content-Type: application/json" \
-d '{"user": "as", "relation":"reader","type":"resource"}' | jq
# user type not found
curl  -X POST 'http://localhost:8080/stores/01GJZNM5ETC4YK34ZZST97R39X/list-objects' \
-H "Content-Type: application/json" \
-d '{"user": "as:asdadsa", "relation":"reader","type":"resource"}' | jq
# user not sent
curl  -X POST 'http://localhost:8080/stores/01GJZNM5ETC4YK34ZZST97R39X/list-objects' \
-H "Content-Type: application/json" \
-d '{"relation":"reader","type":"resource"}' | jq
```

In all these cases, the response was:
```json
{
  "object_ids": []
}
```

Internally, GRPC errors were being (sometimes) thrown and swallowed
```
{"error": "rpc error: code = Code(2009) desc = Invalid input. Make sure you provide a user, object and relation"}
github.com/openfga/openfga/pkg/logger.(*ZapLogger).ErrorWithContext
        /home/runner/work/openfga/openfga/pkg/logger/logger.go:75
github.com/openfga/openfga/server/commands.(*ListObjectsQuery).internalCheck
        /home/runner/work/openfga/openfga/server/commands/list_objects.go:309
github.com/openfga/openfga/server/commands.(*ListObjectsQuery).performChecks.func1
        /home/runner/work/openfga/openfga/server/commands/list_objects.go:277
golang.org/x/sync/errgroup.(*Group).Go.func1
        /home/runner/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75

```

### PR DESCRIPTION
**Pull request by jon-whit**
## Description
Add request validations to ListObjects and cleanup some request validations across packages.

## References
Fixes #365 

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Review by craigpastro — DISMISSED** [review]

**Comment by miparnisari** [comment]
Is this for https://github.com/openfga/openfga/issues/365 ?

**Comment by jon-whit** [comment]
> Is this for #365 ?

Yes, thank you. I'll add it to the references section.

File: server/test/write.go [hunk 1]
```diff
@@ -906,9 +956,15 @@ var writeCommandTests = []writeCommandTest{
...
-			tuple.NewTupleKey("document:budget", "reader", "user:abc"),
+		err: serverErrors.ValidationError(
+			&tuple.InvalidTupleError{
+				Cause:    fmt.Errorf("type 'user' is not an allowed type restriction for 'document#reader'"),
+				TupleKey: tuple.NewTupleKey("document:budget", "reader", "user:abc"),
+			},
 		),
+		// err: serverErrors.InvalidTuple("type 'user' is not an allowed type restriction for 'document#reader'",
+		// 	tuple.NewTupleKey("document:budget", "reader", "user:abc"),
+		// ),
```
Comment (miparnisari) [inline server/test/write.go:967]: remove this?

Comment (jon-whit) [reply server/test/write.go:967]: good catch 👍

Comment (jon-whit) [reply server/test/write.go]: Done.

**Comment by miparnisari** [comment]
Tested locally, looks good

**Review by miparnisari — APPROVED** [review]

**Pull request merged**